test/jest/persistence.spec.ts

```diff
@@ -0,0 +1,155 @@
+import Store, { IStore } from '../../src/store';
+import { IMParticleWebSDKInstance } from '../../src/mp-instance';
+import { SDKInitConfig } from '../../src/sdkRuntimeModels';
+
+describe('Persistence store', () => {
+    let store: IStore;
+    let mockMPInstance: IMParticleWebSDKInstance;
+
+    beforeEach(() => {
+        mockMPInstance = {
+            _Helpers: {
+                createMainStorageName: () => 'mprtcl-v4_ws',
+                createProductStorageName: () => 'mprtcl-prodv4_ws',
+                Validators: { isFunction: () => true },
+                extend: Object.assign,
+            },
+            _NativeSdkHelpers: {
+                isWebviewEnabled: () => false,
+            },
+            _Persistence: {
+                setCookie: jest.fn(),
+                setLocalStorage: jest.fn(),
+                setProductStorage: jest.fn(),
+                update: function () {
+                    if (store.webviewBridgeEnabled) {
+                        return;
+                    }
+                    if (store.getNoFunctional()) {
+                        mockMPInstance._Persistence.setProductStorage();
+                        return;
+                    }
+                    if (store.SDKConfig.useCookieStorage) {
+                        mockMPInstance._Persistence.setCookie();
+                    }
+                    mockMPInstance._Persistence.setLocalStorage();
+                },
+            },
+            Logger: {
+                verbose: () => {},
+                warning: () => {},
+                error: () => {},
+            },
+            Identity: {
+                getCurrentUser: () => ({ getMPID: () => 'mpid' }),
+            },
+        } as unknown as IMParticleWebSDKInstance;
+
+        store = {} as IStore;
+        Store.call(store, {} as SDKInitConfig, mockMPInstance, 'apikey');
+    });
+
+    describe('#noFunctional privacy flag', () => {
+        describe('true', () => {
+            beforeEach(() => {
+                store.setNoFunctional(true);
+                store.webviewBridgeEnabled = false;
+            });
+
+            it('should NOT write to cookie when useCookieStorage is true', () => {
+                store.SDKConfig.useCookieStorage = true;
+                jest.clearAllMocks();
```

Comment:
I think you can add an afterEach block and include jest.clearAllMocks() in that.